src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -770,6 +772,11 @@ void ShenandoahConcurrentGC::op_final_mark() {
     heap->verifier()->verify_roots_no_forwarded();
   }
 
+  {
+    heap->tracer()->report_object_count();
+  }
+
+
```

Comment:
As discussed, move this out of the safepoint.

Reply:
At the call-site, assert that we aren't at a safepoint.
Also add an interlock status on the global klass info table whether it's unpopulated, being populated, populated, being emitted, and emission completed; manage the status  as you start the heap walk/recording, when the walk ends, when you start emitting the events, and when the events have been emitted, adding appropriate assertions to check the states of the machine as you being each phase, and change the state when that phase ends.

This would help you when you do the concurrent ObjectCount events, and also manage potential for interference with GC's and ObjectCountAfterGC. Make sure to do this state recording before you start working on the concurrent ObjectCount recording.